### ISSUE TITLE
Allow for quicker way to view message timestamps

### ISSUE THREAD

**Issue by klinker41** [klinker-apps/messenger-issues#44]
**Device with issue:** Nexus 6P running Android 7.0
**Primary device:** Nexus 6P running Android 7.0
**App version:** 1.1.209

**Information:**

Timestamps are displayed 15 minutes apart right now, which works fine. However, to see every timestamp you have to long click on the message and view details. Some apps (textra, imessage) allow for swiping from the side of the screen to pull up the timestamp list, we could do this as well.

![bwxz8ue](https://cloud.githubusercontent.com/assets/4501322/18142711/6e3cdcce-6f84-11e6-88a9-22b094686ea9.png)
![hpgy4qe](https://cloud.githubusercontent.com/assets/4501322/18142714/72000d2c-6f84-11e6-91d4-65ecc519a7d6.png)

**Comment by klinker24** [comment]
Interesting, that is a neat idea. 

One thing to note: we will have to find a good way to alert users to this ability as well.

**Comment by klinker41** [comment]
We'd also need to figure out how it would work on tablets, the timestamp couldn't come from the side because the list doesn't go all the way to the side.

**Comment by klinker24** [comment]
Maybe instead of swiping from the side, clicking a message could reveal the timestamp below the message? Seems like it could be both easier to implement and easier for a user to understand. 

Although it makes it harder to see the timestamps on all the messages then, which is probably why this was requested in the first place?

**Comment by klinker41** [comment]
I agree with that, would definitely be much easier to implement since it would be on a single message basis. I don't think you'd want to see the timestamps on all messages, just a few probably and not that often either.

**Comment by klinker24** [comment]
For the release milestone, I say that we add the ability to simply single click on messages to see the timestamp. That will give us a baseline and may eliminate this problem completely.

This is another thing we may want to consider an initial tutorial/tooltip for when a user enters a message list for the first time.

**Comment by klinker24** [comment]
There may be a complication. 

While working on [this](https://github.com/klinker41/messenger/commit/27d3c9c1096b7137a411b3395e26c40cfe079272), I found out that the `MotionEvent.ACTION_UP` event must be getting consumed somewhere and never sent to the message `TextView`, although the click listener is working on the image.

I thought that this would probably be an error with `LinkBuilder`, but I disabled everything from that library within the adapter and it still was not passing over those clicks. I am stumped on what the issue could be there.

**Comment by klinker41** [comment]
Is it just because on each of the layouts I've set `android:clickable="true"`? That could be consuming the event before the textview gets it, however why does it work to click on links then? and why do images work to click on?

https://github.com/klinker41/messenger/blob/master/app/src/main/res/layout/message_sent.xml

**Comment by klinker24** [comment]
No, i figured it out, I am working on it. 

It is because you had set it up to use `LinkBuilder`s `LinkConsumableTextView`.

This view actually disallows any touches on the `TextView` so that those touches are forwarded to a `ListView#OnItemClickListener`.

**Comment by klinker24** [comment]
Ok, so the fix:

I did something similar in [PeekView](https://github.com/klinker24/Android-3DTouch-PeekView/blob/master/library/src/main/java/com/klinker/android/peekview/builder/Peek.java). 

When i removed the `LinkConsumableTextView`, it also stopped the ripple effect from happening on the message background. So I made a [ForcedRippleTouchListener](https://github.com/klinker41/messenger/blob/master/app/src/main/java/xyz/klinker/messenger/util/listener/ForcedRippleTouchListener.java) to manually set the ripple to the background.

So now, we were able to use the regular click listener, the long click listener, and it should actually completely fix the issue with the [long click sensitivity](https://github.com/klinker-apps/messenger-issues/issues/49) since we can now receive `MotionEvent.ACTION_UP`.

The one fallback, the ripple itself works great, however, there is a padding at the bottom of each of the message texts that the ripple does not draw on, and i cannot figure out where it is coming from.

But anyways, https://github.com/klinker41/messenger/commit/7f5435d66a35010095dea14a25dbe305ba167259 allows us to click and long click on the individual messages, as well as fixes the sensitivity. The ripple isn't noticable unless you are looking at it, but if you want to take a look @klinker41, you might know better than me how to solve it.

**Comment by klinker41** [comment]
So here it looks like the ripple is showing on the textview background and not the message background and that is where the issue is coming from, correct?

**Comment by klinker41** [comment]
Fixed now with this: https://github.com/klinker41/messenger/commit/7ab5d67a5c6aab9254858691a9d862997eb9495f

Instead of showing the ripple on the child textview, I got that view's parent (the message where the ripple was working previously) and invoked it there. That corrected the padding on the bottom. However, another issue showed up where the ripple was delayed. It seems that the ripple doesn't actually show after setting the state to pressed and enabled until you set it back to blank. Just lowering the timeout fixed this and we get the same effect we originally had now.
